ui/tests/integration/components/http-requests-table-test.js

```diff
@@ -0,0 +1,58 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+const COUNTERS = [
+  {
+    start_time: '2019-05-01T00:00:00Z',
+    total: 50,
+  },
+  {
+    start_time: '2019-04-01T00:00:00Z',
+    total: 45,
+  },
+  {
+    start_time: '2019-03-01T00:00:00Z',
+    total: 55,
+  },
+];
+
+module('Integration | Component | http-requests-table', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+    this.set('counters', COUNTERS);
+  });
+
+  test('it renders', async function(assert) {
+    await render(hbs`<HttpRequestsTable @counters={{counters}}/>`);
+
+    assert.ok(this.element.textContent.trim());
+  });
+
+  test('it does not show Change column with less than one month of data', async function(assert) {
+    const one_month_counter = [
+      {
+        start_time: '2019-05-01T00:00:00Z',
+        total: 50,
+      },
+    ];
+    await render(hbs`<HttpRequestsTable @counters={{one_month_counter}}/>`);
```

Comment:
Does this work without `this.set('one_month_counter', one_month_counter)` ? I think this is probably rendering the empty state in the test which also meets your assertion - you'll likely want to check that it's rendering the table as well as not rendering the change column.

Reply:
it does work -- the table renders headers but with no rows of data. i'll add the extra assertion though, great idea!

meanwhile the `EmptyState` component gets rendered [higher up in the route](https://github.com/hashicorp/vault/pull/6765/files#diff-52841afab7cb52d7de55b7c6780b2d35R17) only if the model does not contain any counters. in other words the `HttpRequestsTable` and `EmptyState` components are never rendered at the same time.

Reply:
The part outside of the loop renders, but counters here will have a length of 0, which is a valid test case, but it looks like this should test the case where counters.length is 1 and for that you'll have to do the `set` before rendering. If not, those bits can be deleted.